tools/stats/print_test_stats.py

```diff
@@ -781,16 +781,21 @@ def assemble_s3_object(
 
 def send_report_to_s3(head_report: Version2Report) -> None:
     job = os.getenv('JOB_BASE_NAME', os.environ.get('CIRCLE_JOB'))
-    # SHARD_NUMBER is specific to GHA jobs, as the shard number would be included in CIRCLE_JOB already
-    shard = os.environ.get('SHARD_NUMBER', '')
     sha1 = os.environ.get('CIRCLE_SHA1')
     branch = os.environ.get('CIRCLE_BRANCH', '')
     now = datetime.datetime.utcnow().isoformat()
+
+    # SHARD_NUMBER and TEST_CONFIG are specific to GHA, as these details would be included in CIRCLE_JOB already
+    shard = os.environ.get('SHARD_NUMBER', '')
+    test_config = os.environ.get('TEST_CONFIG', '')
+
+    job_report_dirname = f'{job}{f"-{test_config}" if test_config else ""}{shard}'
```

Comment:
Nit.
Also I wonder if `warnings.warn` would be appropriate if `TEST_CONFIG` is not defined
```suggestion
    test_config = os.environ.get('TEST_CONFIG')

    job_report_dirname = f'{job}{f"-{test_config}" if test_config is not None else ""}{shard}'
```

Reply:
Hm, would the warnings.warn also apply to SHARD_NUMBER?